File: Lab2/pzpi-22-9-novomlynskyi-viktor-lab2/app/src/main/java/com/example/artguardmobile/data/network/AuthService.kt
package com.example.artguardmobile.data.network

import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.Response

data class LoginRequest(val email: String, val password: String)
data class LoginResponse(val token: String)
data class RegisterRequest(
    val name: String,
    val email: String,
    val password: String
)

data class RegisterResponse(val token: String)

interface AuthService {
    @POST("/api/login")
    suspend fun login(@Body request: LoginRequest): Response<LoginResponse>

    @POST("/api/register")
    suspend fun register(@Body request: RegisterRequest): Response<RegisterResponse>
}

File: Lab3/pzpi-22-9-novomlynskyi-viktor-lab3/src/services/authService.js
import axios from 'axios';

const API_URL = 'http://127.0.0.1:3000/api'; 

export const login = async (email, password) => {
  const response = await axios.post(`${API_URL}/login`, {
    email,
    password,
  });
  return response.data; 
};

export const getMe = async (token) => {
  const response = await axios.get(`${API_URL}/me`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
  return response.data;
};

